launchlib/launcher.go

```diff
@@ -33,41 +33,53 @@ const (
 	ExecPathBlackListRegex = `[^\w.\/_\-]`
 )
 
-type ServiceCommands struct {
-	Primary      *exec.Cmd
-	SubProcesses map[string]*exec.Cmd
+type ServiceCmds struct {
+	Primary             *exec.Cmd
+	PrimaryOutputFile   string
+	SubProcs            map[string]*exec.Cmd
+	SubProcsOutputFiles map[string]string
 }
 
-func CompileCmdsFromConfigFiles(staticConfigFile, customConfigFile string, stdout io.Writer) (ServiceCommands, error) {
-	staticConfig, customConfig, err := GetConfigsFromFiles(staticConfigFile, customConfigFile, stdout)
-	if err != nil {
-		return ServiceCommands{}, err
+func CompileCmdsFromConfig(staticConfig *PrimaryStaticLauncherConfig, customConfig *PrimaryCustomLauncherConfig,
+	primaryStdout io.Writer) (serviceCmds *ServiceCmds, rErr error) {
```

Comment:
Can we replace the multiple instances of calling the parameter `primaryStdout` with just `stdout`.  I think it's pretty confusing given we then create primary output files which themselves have nothing to do with the `stdout`.